lib/puppet/type/file.rb

```diff
@@ -887,6 +887,31 @@ def write(property = nil)
 
   private
 
+  def set_sensitive_properties(sensitive_parameters)
```

Comment:
This is some of the most evil code I've written this year, and I don't think there's a better way to deal with this.